paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -65,20 +65,16 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
     override val bottomSpacer: View by lazy { viewBinding.bottomSpacer }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        val starterArgs = this.starterArgs
+        val validationResult = validateArgs()
+        earlyExitDueToIllegalState = validationResult.isFailure
+
+        super.onCreate(savedInstanceState)
+
+        val starterArgs = validationResult.getOrNull()
         if (starterArgs == null) {
             finish()
             return
```

Comment:
`PaymentOptionsActivity` just finishes without setting an activity result, meaning that the caller won’t ever know about this. Is this something that we want to change?